src/aae_controller.erl

```diff
@@ -24,34 +24,41 @@
             code_change/3]).
 
 -export([aae_start/6,
-            aae_start/7,
-            aae_start/8,
-            aae_nextrebuild/1,
-            aae_put/7,
-            aae_close/1,
-            aae_destroy/1,
-            aae_fetchroot/3,
-            aae_mergeroot/3,
-            aae_fetchbranches/4,
-            aae_mergebranches/4,
-            aae_fetchclocks/5,
-            aae_fetchclocks/7,
-            aae_rebuildtrees/5,
-            aae_rebuildtrees/4,
-            aae_rebuildstore/2,
-            aae_rebuildstore/3,
-            aae_fold/6,
-            aae_fold/8,
-            aae_bucketlist/1,
-            aae_loglevel/2,
-            aae_ping/3,
-            aae_runnerprompt/1
+         aae_start/7,
+         aae_start/8,
+         aae_nextrebuild/1,
+         aae_set_rebuild_schedule/2,
+         aae_get_rebuild_schedule/1,
+         aae_schedulenextrebuild/2,
+         aae_put/7,
+         aae_close/1,
+         aae_destroy/1,
+         aae_fetchroot/3,
+         aae_mergeroot/3,
+         aae_fetchbranches/4,
+         aae_mergebranches/4,
+         aae_fetchclocks/5,
+         aae_fetchclocks/7,
+         aae_rebuildtrees/5,
+         aae_rebuildtrees/4,
+         aae_rebuildstore/2,
+         aae_rebuildstore/3,
+         aae_fold/6,
+         aae_fold/8,
+         aae_bucketlist/1,
+         aae_loglevel/2,
+         aae_ping/3,
+         aae_runnerprompt/1
         ]).
 
 -export([foldobjects_buildtrees/2,
             hash_clocks/2,
             wrapped_splitobjfun/1]).
 
+-export([get_key_store/1,
```

Comment:
Since you are not supposed to make the state of a `gen_server` visible outside the `gen_server` itself, one should not export functions that take that state as input argument.
This is possibly a remainder of an experiment, but I propose to drop those functions and in particular not exporting them.

Reply:
Yeah, it's wip at the moment. I should probably have taken it private until it's ready.